content/blog/2021-10-05-extending-models-with-more-data.md

```diff
@@ -0,0 +1,242 @@
+model improves as we run experiments.
+
+## Updating the dataset with DVC
+
+To add the new cats and dogs dataset to the project, we'll use this DVC command.
+
+```dvc
+dvc get https://github.com/iterative/dataset-registry blog/cats-dogs
+```
+
+This downloads a sample dataset with images of cats and dogs. You can use this
+command to download files or directories that are tracked by DVC or Git. This
+command can be used from anywhere in the file system, as long as DVC is
+installed.
+
+This will make a new directory called `./cats-dogs/data/` that was downloaded
+from the DVC remote and it has images for cats and dogs. The data is in a
+similar format to the ants and bees directories so there aren't any other code
+changes we need to make.
+
+We do need to slowly add in the new data though. For starters, we'll move the
+`val` data for `cats` and `dogs` from the `/cats-dogs/data/` directory to the
+corresponding directory in `data/hymenoptera_data`.
+
+With this new data in place, we can start training our model.
+
+## Running new experiments with generic data
+
+With the updated data, let's run an experiment on the model and see how good the
+results are. To run a new experiment, open your terminal and make sure you have
+a virtual environment enabled. Then run this command:
+
+```dvc
+dvc exp run
+```
+
+Once the training epochs are finished, run the following command.
+
+```dvc
+dvc exp show --no-timestamp --include-metrics step,acc,val_acc,loss,val_loss --include-params lr,momentum
+```
+
+The `--no-timestamp` hides the timestamps from table. The `--includes-metrics`
+option lets us choose which metrics we want to show in the table. The
+`--includes-params` option does the same for hyperparameters. This gives us a
+table that's eaiser to read quickly.
+
+```dvctable
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━┳━━━━━━━━━━┓
+┃ neutral:**Experiment**              ┃ metric:**step** ┃     metric:**acc** ┃ metric:**val_acc** ┃    metric:**loss** ┃ metric:**val_loss** ┃ param:**lr**    ┃ param:**momentum** ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━╇━━━━━━━━━━┩
+│ **workspace**               │    **3** │ **0.86885** │    **0.46** │ **0.31573** │   **3.7067** │ **0.001** │ **0.09**     │
+│ **data-change**             │    **-** │       **-** │       **-** │       **-** │        **-** │ **0.001** │ **0.09**     │
+│ │ ╓ 3b3a2a2 [exp-23593] │    3 │ 0.86885 │    0.46 │ 0.31573 │   3.7067 │ 0.001 │ 0.09     │
+│ │ ╟ 93d015d             │    2 │ 0.83197 │ 0.41333 │ 0.36851 │   3.4259 │ 0.001 │ 0.09     │
+│ │ ╟ d474c42             │    1 │ 0.79918 │ 0.43333 │ 0.46612 │    3.286 │ 0.001 │ 0.09     │
+│ ├─╨ 1582b4b             │    0 │ 0.52869 │    0.39 │ 0.94102 │   2.5967 │ 0.001 │ 0.09     │
+└─────────────────────────┴──────┴─────────┴─────────┴─────────┴──────────┴───────┴──────────┘
+```
+
+You'll notice that the validation accuracy doesn't seem right compared to the
```

Comment:
👍 This is exactly what we should expect because the training metrics are based on bees/ants data and the validation metrics are based on bees/ants/cats/dogs. If we broke down validation metrics by class, they would probably be way better for bees/ants than cats/dogs.

Reply:
Thanks for this explanation! That makes this make more sense to me. I included it in the post because it'll definitely help other people too!